src/app/components/client/PremiumUpsellDialog.tsx

```diff
@@ -95,7 +101,19 @@ function PremiumUpsellDialogContent({
       <div className={styles.productPlans}>
         <TabList
           tabs={tabsData}
-          onSelectionChange={(selectedKey) => setSelectedTab(selectedKey)}
+          onSelectionChange={(selectedKey) => {
+            record("button", "click", {
+              button_id:
+                selectedKey === "monthly"
+                  ? "selected_monthly_plan"
+                  : "selected_yearly_plan",
+              // Mocking the entire session object is too messy in tests:
+              /* c8 ignore next */
+              user_id: session.data?.user.subscriber?.fxa_uid ?? undefined,
+              path: currentPath,
```

Comment:
I think it’s not needed to obtain and pass `path` on each `record` call. For Ga we are already passing it by default and we could change [this line](https://github.com/mozilla/blurts-server/blob/MNTOR-2343-premium-instrumentation/src/app/hooks/useTelemetry.ts#L38) in `useTelemetry` to:
```
void recordGlean(eventModule, event, { ...data, path });
```

Getting and passing `path` in `useGlean` and `useGa` hook is also a good alternative for instances where we might use them standalone.

Reply:
Right, the only reason I didn't do that is because that would mean that every Glean event then needs to define a `path` property - but I realise now that it actually does, so TypeScript shouldn't complain about it. In fact, it will then warn us if we ever add something to `metrics.yml` that _doesn't_ accept a `path`, which is probably a good idea - we'll probably always want to record it.

And in fact, I suppose the same holds for `session_id`. Thus, I added those to `useTelemetry` in ffb396d74.